src/kaleidoscope/hid.cpp

```diff
@@ -358,19 +358,23 @@ void sendKeyboardReport() {
 
   WITH_BOOTKEYBOARD_PROTOCOL {
     if (last_keycode_toggled_on) {
-      BootKeyboard.release(last_keycode_toggled_on);
-      BootKeyboard.sendReport();
-      BootKeyboard.press(last_keycode_toggled_on);
+      if (BootKeyboard.wasKeyPressed(last_keycode_toggled_on)) {
+        BootKeyboard.release(last_keycode_toggled_on);
+        BootKeyboard.sendReport();
+        BootKeyboard.press(last_keycode_toggled_on);
+      }
       last_keycode_toggled_on = 0;
     }
     BootKeyboard.sendReport();
     return;
```

Comment:
One thing I noticed while looking into this is that this `return` statement prevents a keyboard in boot protocol mode from calling `ConsumerControl.sendReport()` below. I don't know if this is intentional, or if it was an oversight. If the latter, it can be easily corrected by using an `else` block instead of this `return`.

Reply:
As far as I remember, when using `BootKeyboard`, we wanted to prevent Consumer and SystemControl keys from being reported. We may need to revisit this stance, and figure out whether to allow them or not (right now, it's not entirely consistent, though we do block the report in the end).

Reply:
If there are approximately no machines that can usefully respond to Consumer Control reports, but can't respond to NKRO keyboards, then that makes perfect sense.

Reply:
The most often used case for BootKeyboard are BIOSes and encryption prompts, consumer and systemcontrol doesn't make much sense there, and neither handles them. There are some OSes that do not implement NKRO, I need to check if they can do anything with consumer or systemcontrol keys (FreeBSD comes to mind as the most notable OS without NKRO).